### DISCUSSION TITLE
Wrong default value is displayed for a string Enum option

### DISCUSSION THREAD

**Discussion by avoskresensky** [fastapi/typer#703] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional, Annotated

import logging
import typer as t
from enum import Enum

root = t.Typer()

class LogLevel(str, Enum):
    """LogLevel enumeration.
    """
    DEBUG = 'DEBUG'
    INFO = 'INFO'
    WARNING = 'WARNING'
    ERROR = 'ERROR'
    CRITICAL = 'CRITICAL'

@root.command()
def _root(log_level: Annotated[Optional[LogLevel], t.Option(case_sensitive=False)] = LogLevel.WARNING):
    if log_level:
        logger = logging.getLogger()
        logger.setLevel(log_level.value)
```

### Description

Here's what I get in --help for the above code:

```
--log-level                 [DEBUG|INFO|WARNING|ERROR|CRITICAL]  [default: LogLevel.WARNING] 
```

The displayed default value uses the Python enum notation.
I'm expecting to see a user friendly "default: WARNING" instead.

### Operating System

Windows

### Typer Version

0.9.0

### Python Version

3.11.6

**Comment by svlandeg** [comment, marked as answer]
Hi! Thanks for the report. This behaviour should be fixed by PR #473 😎
